src/mots/config.py

```diff
@@ -201,7 +202,23 @@ def clean(file_config: FileConfig, write: bool = True):
     bmo_data = get_bmo_data(people)
     updated_people = People(people, bmo_data)
     logger.debug("Updating people configuration based on BMO data...")
-    file_config.config["people"] = updated_people.serialized
+
+    # If refresh is True, then we should do this. Otherwise, we should pick out
+    # pre-existing users and not touch those.
+    if refresh:
+        logger.debug("Refreshing all people...")
+        file_config.config["people"] = updated_people.serialized
+    else:
+        logger.warning("Only synchronizing new people with Bugzilla.")
+        updated_people_dict = {p["bmo_id"]: p for p in updated_people.serialized}
+        people_dict = {p["bmo_id"]: p for p in people}
+
+        additions = {
+            p: v for p, v in updated_people_dict.items() if "nick" not in people_dict[p]
+        }
+        parsed_people = [p for p in people if p["bmo_id"] not in additions]
+        parsed_people += additions.values()
```

Comment:
There is a bug here that causes indices to get corrupt. New people should be inserted in their original location in the list instead.